Author: Mikhail Gusarov <dottedmag@dottedmag.net>
Description: [PATCH] Return non-zero code from 'put' command if some files failed
 to upload
Forwarded: https://github.com/s3tools/s3cmd/pull/507
Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=669002
---
 s3cmd | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: s3cmd/s3cmd
===================================================================
--- s3cmd.orig/s3cmd	2015-03-18 11:54:56.630549438 +0100
+++ s3cmd/s3cmd	2015-03-18 11:54:56.626549422 +0100
@@ -363,6 +363,7 @@
         return EX_OK
 
     seq = 0
+    ret = EX_OK
     for key in local_list:
         seq += 1
 
@@ -382,9 +383,11 @@
             response = s3.object_put(full_name, uri_final, extra_headers, extra_label = seq_label)
         except S3UploadError, e:
             error(u"Upload of '%s' failed too many times. Skipping that file." % full_name_orig)
+            ret = EX_SOMEFAILED
             continue
         except InvalidFileError, e:
             warning(u"File can not be uploaded: %s" % e)
+            ret = EX_SOMEFAILED
             continue
         if response is not None:
             speed_fmt = formatSize(response["speed"], human_readable = True, floating_point = True)
@@ -398,7 +401,7 @@
         if Config().encrypt and full_name != full_name_orig:
             debug(u"Removing temporary encrypted file: %s" % unicodise(full_name))
             os.remove(full_name)
-    return EX_OK
+    return ret
 
 def cmd_object_get(args):
     cfg = Config()
